util/log/src/main/scala/sbt/ConsoleLogger.scala

```diff
@@ -35,6 +35,19 @@ object ConsoleLogger {
   def isEscapeTerminator(c: Char): Boolean =
     c >= '@' && c <= '~'
 
+  /**
+   * Test if the character AFTER an ESC is the ANSI CSI, which means more than two characters until terminatioin.
+   */
+  def isCSI(c: Char): Boolean = c == '['
+
+  /**
+   * Tests whether or not a character needs to immediately terminate the ANSI sequence.
+   */
+  def isSimpleEscapeTerminator(c: Char): Boolean = {
+    val n = c.toInt
+    (n >= 64) && (n <= 95)
```

Comment:
Yeah, all good points. I'll add the references and more description of why.

TBH - i'm really not sure what the "known escapes" are from in the tests, still.  They're definitely not ANSI, but there must be a reason we try to escape them.